src/services/navigator/index.ts

```diff
@@ -0,0 +1,319 @@
+    const state = navigation.getState();
+
+    if (state.type === Types.NAVIGATOR_TYPE.STACK && state.index > 0) {
+      const prev = state.routes[state.index - 1];
+
+      navigation.dispatch({
+        ...Imports.CommonActions.setParams({
+          ...routeParams,
+        }),
+        source: prev.key,
+        target: state.key,
+      });
+    }
+    navigation.goBack();
+  }
+
+  /**
+   * Prepare and send the request
+   */
+  sendRequest = (
+    action: TypesLegacy.NavAction,
+    routeParams: TypesLegacy.NavigationRouteParams,
+  ) => {
+    let { navigation } = this.props;
+    let routeId: string | undefined;
+    let params:
+      | Types.NavigationNavigateParams
+      | TypesLegacy.NavigationRouteParams
+      | undefined;
+    let navAction: TypesLegacy.NavAction = action;
+
+    if (routeParams) {
+      // The push action is used for urls which are not associated with a route
+      // See use of `this.getRouteKey(url);` in `hyperview/src/services/navigation`
+      if (routeParams.url) {
+        if (
+          navAction === TypesLegacy.NAV_ACTIONS.PUSH &&
+          Helpers.isUrlFragment(routeParams.url)
+        ) {
+          navAction = TypesLegacy.NAV_ACTIONS.NAVIGATE;
+        }
+      }
+
+      const [
+        requestNavigation,
+        requestRouteId,
+        requestParams,
+      ] = this.buildRequest(navAction, routeParams || {});
+
+      navigation = requestNavigation;
+      routeId = requestRouteId;
+      params = requestParams;
+    }
+
+    if (!navigation) {
+      if (routeParams.targetId) {
+        console.warn(
+          `No navigation found for target '${routeParams.targetId}'`,
+        );
+      }
+      return;
+    }
+
+    switch (navAction) {
+      case TypesLegacy.NAV_ACTIONS.BACK:
+        if (routeParams) {
+          this.routeBackRequest(navigation, routeParams);
+        } else {
+          navigation.goBack();
+        }
+        break;
+      case TypesLegacy.NAV_ACTIONS.CLOSE:
+        navigation.goBack();
+        break;
+      case TypesLegacy.NAV_ACTIONS.NAVIGATE:
+        if (routeId) {
+          navigation.dispatch(Imports.CommonActions.navigate(routeId, params));
+        }
+        break;
+      case TypesLegacy.NAV_ACTIONS.NEW:
+        if (routeId) {
+          navigation.dispatch(Imports.CommonActions.navigate(routeId, params));
+        }
+        break;
+      case TypesLegacy.NAV_ACTIONS.PUSH:
+        if (routeId) {
+          navigation.dispatch(Imports.StackActions.push(routeId, params));
+        }
+        break;
+      default:
+    }
+  };
+
+  back = (params: TypesLegacy.NavigationRouteParams) => {
+    this.sendRequest(TypesLegacy.NAV_ACTIONS.BACK, params);
+  };
+
+  closeModal = (params: TypesLegacy.NavigationRouteParams) => {
+    this.sendRequest(TypesLegacy.NAV_ACTIONS.CLOSE, params);
+  };
+
+  // eslint-disable-next-line @typescript-eslint/no-unused-vars, no-unused-vars
+  navigate = (params: TypesLegacy.NavigationRouteParams, _: string) => {
```

Comment:
do we need to define the second argument if it's not used?

Reply:
Fixed.